src/Plugins/ComplexCore/test/CopyFeatureArrayToElementArrayTest.cpp

```diff
@@ -0,0 +1,117 @@
+  REQUIRE(avgTempValuePtr != nullptr);
+  DataArray<T>& avgTempValue = *avgTempValuePtr;
+
+  for(int i = 0; i < 3; i++)
+  {
+    avgTempValue[i] = static_cast<T>(0);
+  }
+
+  // Create filter and set arguments
+  CopyFeatureArrayToElementArray filter;
+  Arguments args;
+
+  args.insertOrAssign(CopyFeatureArrayToElementArray::k_SelectedFeatureArrayPath_Key, std::make_any<DataPath>(DataPath({k_FeatureDataArrayName})));
+  args.insertOrAssign(CopyFeatureArrayToElementArray::k_FeatureIdsArrayPath_Key, std::make_any<DataPath>(DataPath({k_CellFeatureIdsArrayName})));
+  args.insertOrAssign(CopyFeatureArrayToElementArray::k_CreatedArrayName_Key, std::make_any<DataPath>(DataPath({k_CellTempArrayName})));
+
+  // Preflight the filter
+  auto preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.valid());
+
+  // Execute the filter
+  auto executeResult = filter.execute(ds, args);
+  REQUIRE(executeResult.result.valid());
+
+  // Check the filter results
+  DataArray<T>& createdElementArray = ds.getDataRefAs<DataArray<T>>(DataPath({k_CellTempArrayName}));
+  for(usize i = 0; i < createdElementArray.getNumberOfTuples(); i++)
+  {
+    int32 featureId = cellFeatureIds[i];
+    T value = createdElementArray[i];
+    T featureValue = avgTempValue[featureId];
+    REQUIRE(value == featureValue);
+  }
+}
+} // namespace
+
+TEST_CASE("Core::CopyFeatureArrayToElementArray: Instantiation and Parameter Check", "[Core][CopyFeatureArrayToElementArray]")
+{
+  // Instantiate the filter, a DataStructure object and an Arguments Object
+  CopyFeatureArrayToElementArray filter;
+  DataStructure ds;
+  Arguments args;
+
+  // Create default Parameters for the filter.
+  args.insertOrAssign(CopyFeatureArrayToElementArray::k_SelectedFeatureArrayPath_Key, std::make_any<DataPath>(DataPath{}));
+  args.insertOrAssign(CopyFeatureArrayToElementArray::k_FeatureIdsArrayPath_Key, std::make_any<DataPath>(DataPath{}));
+  args.insertOrAssign(CopyFeatureArrayToElementArray::k_CreatedArrayName_Key, std::make_any<DataPath>(DataPath{}));
+
+  // Preflight the filter and check result
+  auto preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.invalid());
+  REQUIRE(preflightResult.outputActions.errors().size() == 3);
+  for(const Error& err : preflightResult.outputActions.errors())
+  {
+    REQUIRE(err.code == k_EmptyArrayPath);
+  }
+
+  // Execute the filter and check the result
+  auto executeResult = filter.execute(ds, args);
+  REQUIRE(executeResult.result.invalid());
+  REQUIRE(executeResult.result.errors().size() == 3);
+  for(const Error& err : executeResult.result.errors())
+  {
+    REQUIRE(err.code == k_EmptyArrayPath);
+  }
+}
+
+TEST_CASE("Core::CopyFeatureArrayToElementArray: Valid filter execution", "[Core][CopyFeatureArrayToElementArray]")
+{
+  TestCopyFeatureArrayToElementArrayForType<uint8>();
```

Comment:
I haven't used this feature before, but it might be worth looking into Catch2's `TEMPLATE_TEST_CASE` macro.

https://github.com/catchorg/Catch2/blob/devel/docs/test-cases-and-sections.md#type-parametrised-test-cases